src-java/server42/server42-control-server-stub/src/main/java/org/openkilda/server42/control/serverstub/StatsServer.java

```diff
@@ -114,15 +115,23 @@ void init() {
      */
     public synchronized void addFlow(Flow flow) {
 
+        FlowKey flowKey = FlowKey.fromFlow(flow);
+
+        if (flows.containsKey(flowKey) && flows.get(flowKey).getFlowHashCode() == flow.getHashCode()) {
```

Comment:
Need to be careful with == operator. The operator works here only because flow.getHashCode() returns a primitive. If it returns wrapper class as getFlowHashCode() does, then the operator will fail.